File: gestion-tickets/src/app/DTOs/ClientDto.ts
import { ProjetMini } from "./ProjetMini";

export interface ClientDto {
  id: number;
  email: string;
  firstName: string;
  lastName: string;
  numTelephone: string;
  actif: boolean;
  initialPassword: string;
  paysId: number;
  pays: string; 
  societeId: number;
  societe: {
    id: number;
    nom: string;
  };
  createdAt: string;
  updatedAt: string;
  deletedAt?: string;
  token?: string;
  projets: ProjetMini[];
}

File: GestionTicketsAPI/DTOs/ClientDto.cs
using System;

namespace GestionTicketsAPI.DTOs;

public class ClientDto
    {
        public int Id { get; set; }
        public required string Email { get; set; }
        public required string FirstName { get; set; }
        public required string LastName { get; set; }
        public required string NumTelephone { get; set; }
        public int PaysId { get; set; }
        public required string Pays { get; set; }
        public required bool Actif { get; set; }
        public string PhotoUrl { get; set; }
        public string? Token { get; set; }
        public string InitialPassword { get; set; } = string.Empty;
        public int SocieteId { get; set; }  
        public SocieteDto Societe { get; set; } = default!;
        public ICollection<ProjetMiniDto> Projets { get; set; } = new List<ProjetMiniDto>();
    }
